### PR TITLE
typecheck createDataFrame from emptyRDD & fillna methods

### PR DESCRIPTION
**Pull request by oliverw1**
emptyRDDs are sometimes used in tests and they currently raise a typecheck error.

The `pyspark.sql.DataFrame.fillna` method also raises an error when passing a single string for the subset keyword. While fixing this, I saw that tuples are currently also not passed by pyspark-stubs, even though the public API does allow it. That makes this issue similar to some of my previous PRs here (and in general, an issue with the Python community for using mutable lists everywhere, when tuples (or sets, or any other iterable) would suffice /pet peeve).

The last commit (7d588a5) is merely an insurance against future possible issues on e.g. files with spaces in the filename. It should also run faster, though on these small listings, it won't be noticeable.

Looking forward to hearing your thoughts, especially on `RowLike` now being a type alias for None.

### CODE REVIEW

File: third_party/3/pyspark/sql/dataframe.pyi [hunk 1]
```diff
@@ -180,7 +180,9 @@ class DataFrame(PandasMapOpsMixin, PandasConversionMixin):
     ) -> DataFrame: ...
     @overload
     def fillna(
-        self, value: LiteralType, subset: Optional[List[str]] = ...
+        self,
+        value: LiteralType,
+        subset: Optional[Union[str, Tuple[str], List[str]]] = ...,
```
Comment (zero323) [inline third_party/3/pyspark/sql/dataframe.pyi:185]: Shouldn't it be `Tuple[str, ...]`?

**Comment by zero323** [comment]
Thanks @oliverw1. In general these look good for me, with exception of the question raised above.

File: third_party/3/pyspark/sql/dataframe.pyi (hunk 1 above)
Comment (zero323) [reply third_party/3/pyspark/sql/dataframe.pyi:185]: In other words, could you please test it with

```python
df.fillna(value=1, subset=("id1", "id2"))
```

?

TIA.

File: third_party/3/pyspark/sql/_typing.pyi [hunk 2]
```diff
@@ -32,7 +32,7 @@ AtomicDataTypeOrString = Union[pyspark.sql.types.AtomicType, str]
 DataTypeOrString = Union[pyspark.sql.types.DataType, str]
 OptionalPrimitiveType = Optional[PrimitiveType]
 
-RowLike = TypeVar("RowLike", List[Any], Tuple[Any, ...], pyspark.sql.types.Row)
+RowLike = TypeVar("RowLike", List[Any], Tuple[Any, ...], pyspark.sql.types.Row, None)
```
Comment (zero323) [inline third_party/3/pyspark/sql/_typing.pyi:35]: After some thought I don't think this the way to go. Instead, annotation `emptyRDD` should be changed to

```python
    def emptyRDD(self) -> RDD[Any]: ...
```

Comment (oliverw2) [reply third_party/3/pyspark/sql/_typing.pyi]: You're right, that seems like a better way to do it. It's more consistent, as calling `collect` on any RDD results in a list of a priori undefined type (unlike Scala's typed Datasets), so Any is a better fit. Fixing…

**Comment by zero323** [comment]
Thanks.

Since these are three separate features, I'd prefer not squash the whole thing ‒ if we're going to include this in a single PR it would be great if you could squash related commits on your side (i.e. 5e3a71cfe2aff0fe3431e550a8d6d0dc1ed40803 with fe8b69ce7247d0ad9e85d70680822ed8bef5de75).

**Comment by oliverw1** [comment]
Agreed, makes sense. I rebased the changes and force-pushed the branch.

**Review by zero323 — APPROVED** [review]

**Pull request merged**